optuna/_gp/search_space.py

```diff
@@ -92,8 +96,14 @@ def sample_normalized_params(
     scale_types = search_space.scale_types
     bounds = search_space.bounds
     steps = search_space.steps
-    qmc_engine = qmc.Sobol(dim, scramble=True, seed=rng.randint(np.iinfo(np.int32).max))
-    param_values = qmc_engine.random(n)
+
+    # Sobol engine likely shares its internal state among threads.
+    # Without threading.Lock, ValueError exceptions are raised in Sobol engine as discussed in
+    # https://github.com/optuna/optunahub-registry/pull/168#pullrequestreview-2404054969
+    with _threading_lock:
+        qmc_engine = qmc.Sobol(dim, scramble=True, seed=rng.randint(np.iinfo(np.int32).max))
+        param_values = qmc_engine.random(n)
```

Comment:
I'm not sure about the behavior of `Sobol`, but it works without the lock for the sampling.

```suggestion
    param_values = qmc_engine.random(n)
```

Reply:
Thank you for your review.

I'm also unsure about the behavior. Considering the Sobol's sampling works faster, it would be safer to lock both two lines for now. What do you think?

Reply:
We discussed offline and decided to apply this suggestion.